Scripts/Actions/DebugMenuActionsPage.cs

```diff
@@ -0,0 +1,185 @@
+				var part = parts[i];
+				if (currentNode.ContainsChild(part))
+				{
+					currentNode = currentNode.GetChild(part);
+				}
+				else
+				{
+					currentNode =
+						isLast ?
+							currentNode.AddButton(part, action, hideDebugMenuOnClick) :
+							currentNode.AddFolder(part);
+				}
+			}
+		}
+
+		/// <summary>
+		/// Removes a button from the debug menu.
+		/// </summary>
+		/// <param name="path">The path of of the button to remove</param>
+		public void RemoveButton(string path)
+		{
+			if (string.IsNullOrEmpty(path)) throw new ArgumentNullException(nameof(path));
+
+			if (!ButtonPathExists(path))
+			{
+				throw new Exception($"Cannot remove the button {path}. It does not exist.");
+			}
+
+			var parts = GetParts(path);
+			var nodes = new List<DebugMenuItemNode>();
+			var currentNode = rootNode;
+			nodes.Add(currentNode);
+			foreach (var part in parts)
+			{
+				currentNode = currentNode.GetChild(part);
+				nodes.Add(currentNode);
+			}
+
+			for (var i = nodes.Count - 2; i >= 0; i--)
+			{
+				var parent = nodes[i];
+				var child = nodes[i + 1];
+				if (child.ChildCount == 0)
+				{
+					parent.RemoveChild(child.Name);
+				}
+			}
+		}
+
+		private void SetupPage(DebugMenuItemNode node)
+		{
+			currentPageNode = node;
+			foreach (Transform child in actionButtonTemplate.transform.parent)
+			{
+				if (child != actionButtonTemplate.transform)
+				{
+					Destroy(child.gameObject);
+				}
+			}
+
+			GameObject firstChild = null;
+
+			if (node.Parent != null)
+			{
+				var backButton = Instantiate(actionButtonTemplate, actionButtonTemplate.transform.parent, false);
+				backButton.GetComponentInChildren<Text>().text = $"{backCharacter} Back";
+				backButton.onClick.AddListener(() => { SetupPage(node.Parent); });
+				backButton.gameObject.SetActive(true);
+
+				firstChild = backButton.gameObject;
+			}
+
+			foreach (var button in node.Children.Values)
+			{
+				var actionButton = Instantiate(actionButtonTemplate, actionButtonTemplate.transform.parent, false);
+				if (button.IsFolder)
+				{
+					actionButton.onClick.AddListener(() => { SetupPage(button); });
+				}
+				else
+				{
+					actionButton.onClick.AddListener(() =>
+					{
+						button.Action.Invoke();
+						if (button.HideOnClick) debugMenu.Hide();
+					});
+				}
+
+				var label = actionButton.GetComponentInChildren<Text>();
+				label.text = button.Name + (button.IsFolder ? $" {folderArrowCharacter}" : "");
+				actionButton.gameObject.SetActive(true);
+
+				firstChild = (firstChild != null) ? firstChild : actionButton.gameObject;
+			}
+
+			if (debugMenu.UseNavigation && firstChild != null)
+			{
+				EventSystem.current.SetSelectedGameObject(firstChild);
+			}
+		}
+
+		private bool ButtonPathExists(string path)
+		{
+			var parts = GetParts(path);
+			var currentNode = rootNode;
+			foreach (var part in parts)
+			{
+				if (currentNode.ContainsChild(part))
+				{
+					currentNode = currentNode.GetChild(part);
```

Comment:
Is it faster if you just `GetChild` then do a null check, rather than check `ContainsChild` first then `GetChild`?